### PR TITLE
Add module 6 examples

### PR DESCRIPTION
**Pull request by kjendrzyca**
First draft

### CODE REVIEW

File: module-6/lsp-example.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+}
+Avatar.propTypes = {
+  pictureData: PropTypes.shape({
+    url: PropTypes.string.isRequired,
+    size: PropTypes.number.isRequired,
+  })
+}
+NavigationPicture.propTypes = {
+  pictureData: PropTypes.shape({
+    url: PropTypes.shape({ // <- WTF
```
Comment (tarnas14) [inline module-6/lsp-example.js]: co WTF?

File: module-6/ocp-example.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    <div>
+      {numeralJs(adjustedFormat)(number)}
+    </div>
+  )
+}
+
+// !!! Requirement change: if the `display negatives in parenthesis` in the settings is set !!!
+// !!! then negatives should be displayed in parenthesis !!!
+// Before: -10000.000
+// After: (10000.000)
```
Comment (tarnas14) [inline module-6/ocp-example.js]: to jest wtf

File: module-6/ocp-example.js [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+function withAdjustedFormatting(ComponentToWrap) {
+  function LocalizedFormatter({ displayNegativesInParenthesis, ...restProps }) {
+    const locationAdjustedFormat = format => checkIfUS() ? `0,${format}` : format // 0,0.000
+    const settingsAdjustedFormat = format => displayNegativesInParenthesis
+      ? `(${format})` // (0,0.000)
+      : format
+
+    const adjustFormatting = format => settingsAdjustedFormat(locationAdjustedFormat(format))
+
+    return <ComponentToWrap adjustFormatting={adjustFormatting} {...restProps} />
```
Comment (tarnas14) [inline module-6/ocp-example.js]: ja bym zrobił `{...props} adjustFormatting={adjustFormatting}` żeby przypadkiem nikt nie napisał tego (ale to paranoja)

File: module-6/ocp-example.js (hunk 2 above)
Comment (kjendrzyca) [reply module-6/ocp-example.js]: https://www.accountingcoach.com/blog/use-of-parentheses

File: module-6/lsp-example.js (hunk 1 above)
Comment (kjendrzyca) [reply module-6/lsp-example.js]: The idea is that in the other components the `url` is a string, here it's an object. The simplest version of LSP tells that you shouldn't surprise people, here we are breaking the rule a little bit.

File: module-6/ocp-example.js (hunk 3 above)
Comment (kjendrzyca) [reply module-6/ocp-example.js]: fixed

File: module-6/ocp-example.js (hunk 2 above)
Comment (tarnas14) [reply module-6/ocp-example.js]: what the actual fuck

**Pull request merged**